File: 240620/격자 반대로 채우기/grid-anti-fill.py
N = int(input())
board = [[0] * N for _ in range(N)]

num = 0
if N % 2 == 0:
    for j in range(N - 1, -1, -1):
        if j % 2 == 1:
            for i in range(N - 1, -1, -1):
                num += 1
                board[i][j] = num
        else:
            for i in range(N):
                num += 1
                board[i][j] = num
else:
    for j in range(N - 1, -1, -1):
        if j % 2 == 0:
            for i in range(N - 1, -1, -1):
                num += 1
                board[i][j] = num
        else:
            for i in range(N):
                num += 1
                board[i][j] = num

for row in board:
    print(*row)

# # 모범답안
# # 변수 선언 및 입력
# n = int(input())
# answer = [
#     [0 for _ in range(n)]
#     for _ in range(n)
# ]
# count = 1

# # 격자를 채워줍니다.
# for col in range(n - 1, -1, -1):
#     if (n - 1 - col) % 2 == 0:
#         # 이 케이스에는 아래에서 위로 값을 채워줍니다.
#         for row in range(n - 1, -1, -1):
#             answer[row][col] = count
#             count += 1
#     else:
#         # 이 케이스에는 위에서 아래로 값을 채워줍니다.
#         for row in range(n):
#             answer[row][col] = count
#             count += 1

# # 출력:
# for row in answer:
#     for elem in row:
#         print(elem, end=" ")
#     print()

File: 251125/격자 반대로 채우기/grid-anti-fill.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int N;
    cin >> N;

    vector<vector<int>> board(N, vector<int>(N, 0));

    int num = 0;

    if (N % 2 == 0) 
    {
        for (int j = N - 1; j >= 0; --j) 
        {
            if (j % 2 == 1) 
            {                
                for (int i = N - 1; i >= 0; --i) 
                {
                    num++;
                    board[i][j] = num;
                }
            } 
            else 
            {
                for (int i = 0; i < N; ++i) 
                {
                    num++;
                    board[i][j] = num;
                }
            }
        }
    } 
    else 
    {
        for (int j = N - 1; j >= 0; --j) 
        {
            if (j % 2 == 0) 
            {
                for (int i = N - 1; i >= 0; --i) 
                {
                    num++;
                    board[i][j] = num;
                }
            } 
            else 
            {
                for (int i = 0; i < N; ++i) 
                {
                    num++;
                    board[i][j] = num;
                }
            }
        }
    }

    for (int i = 0; i < N; ++i) 
    {
        for (int j = 0; j < N; ++j) 
        {
            cout << board[i][j];
            if (j + 1 < N) cout << ' ';
        }
        cout << '\n';
    }

    return 0;
}
